Distributed tracing 2.9 (#62410)
openshift/openshift-docs: modules/distr-tracing-tempo-deploy-default.adoc

// Module included in the following assemblies:
//
// * distr_tracing_tempo/distr-tracing-tempo-configuring.adoc

:_content-type: PROCEDURE
[id="distr-tracing-tempo-deploy-default_{context}"]
= Deploying the {DTShortName} default strategy from the web console

> The custom resource definition (CRD) defines the configuration used when you deploy an instance of {DTProductName}. The default CR is named `tempo-all-in-one-inmemory` and it is configured with minimal resources to ensure that you can successfully install it on a default {product-title} installation. You can use this default configuration to create a {TempoName} instance that uses the `AllInOne` deployment strategy, or you can define your own custom resource file.

Reviewer: ```suggestion
The custom resource definition (CRD) defines the configuration used when you deploy an instance of {DTProductName}. The default CR is configured with minimal resources to ensure that you can successfully install it on a default {product-title} installation. You can use this default configuration to create a basic {TempoName} instance, or you can define your own custom resource file.
```

[NOTE]
====
In-memory storage is not persistent. If the Tempo pod shuts down, restarts, or is replaced, your trace data will be lost. For persistent storage, you must use the `production` or `streaming` strategies, which use Elasticsearch as the default storage.

> ====

Reviewer: This can be removed

Author: Outdated. Resolving this thread.

.Prerequisites

* The {TempoName} Operator has been installed.
* You have reviewed the instructions for how to customize the deployment.

> * You have access to the cluster as a user with the `cluster-admin` role.

Reviewer: ```suggestion
* You have access to the cluster as a user with the `cluster-admin` role.
* Supported Object Store (AWS S3, Google Cloud Storage, Azure, Minio, OpenShift Data Foundation)
```

.Procedure

. Log in to the {product-title} web console as a user with the `cluster-admin` role.

. Create a new project, for example `tracing-system`.
+
[NOTE]
====

> If you are installing as part of Service Mesh, the {DTShortName} resources must be installed in the same namespace as the `ServiceMeshControlPlane` resource, for example `istio-system`.

Reviewer 2: Should we remove this note from tempo docs? I guess yes

====
+
.. Navigate to *Home* -> *Projects*.

.. Click *Create Project*.

.. Enter `tracing-system` in the *Name* field.

.. Click *Create*.

. Navigate to *Operators* -> *Installed Operators*.

. If necessary, select `tracing-system` from the *Project* menu. You may have to wait a few moments for the Operators to be copied to the new project.

. Click the {TempoName} Operator. On the *Details* tab, under *Provided APIs*, the Operator provides a single link.

> . Under *Tempo*, click *Create Instance*.

Reviewer: We need to specify, at some point, the creation of the secret for the storage

Author: Outdated. Resolving this thread.

. On the *Create Tempo* page, to install using the defaults, click *Create* to create the {TempoShortName} instance.

> . On the *Tempos* page, click the name of the {TempoShortName} instance, for example, `tempo-all-in-one-inmemory`.

Reviewer: ```suggestion
. On the *Tempos* page, click the name of the {TempoShortName} instance, for example, `example`.
```

. On the *Tempo Details* page, click the *Resources* tab. Wait until the pod has a status of "Running" before continuing.

[id="distr-tracing-deploy-default-cli_{context}"]
== Deploying the {DTShortName} default strategy from the CLI

Follow this procedure to create an instance of {TempoShortName} from the command line.

.Prerequisites

* The {TempoName} Operator has been installed and verified.
* You have reviewed the instructions for how to customize the deployment.
* You have access to the OpenShift CLI (`oc`) that matches your {product-title} version.
* You have access to the cluster as a user with the `cluster-admin` role.

.Procedure

. Log in to the {product-title} CLI as a user with the `cluster-admin` role.
+
[source,terminal]
----
$ oc login --username=<NAMEOFUSER> https://<HOSTNAME>:8443
----

. Create a new project named `tracing-system`.
+
[source,terminal]
----
$ oc new-project tracing-system
----

> 

Reviewer: ```suggestion

. Create a Secret YAML file to specify the secret for your object storage bucket. For instance, if you are using AWS S3, create a file that uses the `access_key_id` and `access_key_secret` fields to specify your AWS credentials and bucketnames, endpoint and region to define the object storage location. For example:

[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: minio-secret
stringData:
  endpoint: http://minio.minio.svc:9000
  bucket: tempo
  access_key_id: tempo
  access_key_secret: supersecret
type: Opaque
----

```

. Create a custom resource file named `tempo.yaml` that contains the following text:
+

> .Example tempo-all-in-one.yaml

Reviewer: ```suggestion
.Example example.yaml
```

> [source,yaml]

Reviewer: We need to review this CR. Some required steps are missing. And the `apiVersion` is wrong

----
apiVersion: tempotracing.io/v1
kind: Tempo
metadata:

>   name: tempo-all-in-one-inmemory

Reviewer: ```suggestion
  name: example
```

----

. Run the following command to deploy {TempoShortName}:
+
[source,terminal]
----
$ oc create -n tracing-system -f tempo.yaml
----

. Run the following command to watch the progress of the pods during the installation process:
+
[source,terminal]
----
$ oc get pods -n tracing-system -w
----
+
After the installation process has completed, you should see output similar to the following example:
+
[source,terminal]
----
NAME                                         READY   STATUS    RESTARTS   AGE

> tempo-all-in-one-inmemory-cdff7897b-qhfdx   2/2     Running   0          24s

Reviewer: Review this

[NOTE]
====
In-memory storage is not persistent. If the Tempo pod shuts down, restarts, or is replaced, your trace data will be lost. For persistent storage, you must use the `production` or `streaming` strategies, which use Elasticsearch as the default storage.
====

.Prerequisites

* The {TempoName} Operator has been installed.
* You have reviewed the instructions for how to customize the deployment.
* You have access to the cluster as a user with the `cluster-admin` role.
* Supported Object Store (AWS S3, Google Cloud Storage, Azure, Minio, OpenShift Data Foundation)

.Procedure

. Log in to the {product-title} web console as a user with the `cluster-admin` role.

. Create a new project, for example `tracing-system`.
+
[NOTE]
====
If you are installing as part of Service Mesh, the {DTShortName} resources must be installed in the same namespace as the `ServiceMeshControlPlane` resource, for example `istio-system`.
====
+

> .. Navigate to *Home* -> *Projects*.

Reviewer 2: I would add a note here like this:

[NOTE]
====
Before the creation of the tempoStack instance  you need to create a secret that will contain the storage backend configuration.
====

.. Click *Create Project*.

.. Enter `tracing-system` in the *Name* field.

.. Click *Create*.

. Navigate to *Operators* -> *Installed Operators*.

. If necessary, select `tracing-system` from the *Project* menu. You may have to wait a few moments for the Operators to be copied to the new project.

. Click the {TempoName} Operator. On the *Details* tab, under *Provided APIs*, the Operator provides a single link.

. Under *Tempo*, click *Create Instance*.

> 

Reviewer 2: . Need to configure the object storage backend, create form options select *Object Storage* -> *Object Storage Secret* -> *Object Storage Secret Name* and select a secret from the list, If you haven't created a secret yet you need to create one depending on the backend storage you use.

. Under *Object Storage* -> *Object Storage Secret* select the object storage type, could be s3, azure or gcp

Follow this procedure to create an instance of {TempoShortName} from the command line.

.Prerequisites

* The {TempoName} Operator has been installed and verified.
* You have reviewed the instructions for how to customize the deployment.
* You have access to the OpenShift CLI (`oc`) that matches your {product-title} version.
* You have access to the cluster as a user with the `cluster-admin` role.

.Procedure

. Log in to the {product-title} CLI as a user with the `cluster-admin` role.
+
[source,terminal]
----
$ oc login --username=<NAMEOFUSER> https://<HOSTNAME>:8443
----

. Create a new project named `tracing-system`.
+
[source,terminal]
----
$ oc new-project tracing-system
----

. Create a Secret YAML file to specify the secret for your object storage bucket. For instance, if you are using AWS S3, create a file that uses the `access_key_id` and `access_key_secret` fields to specify your AWS credentials and bucketnames, endpoint and region to define the object storage location. For example:

[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: minio-secret
stringData:
  endpoint: http://minio.minio.svc:9000
  bucket: tempo
  access_key_id: tempo

>   access_key_secret: supersecret

Author: Would it be possible to add some more realistic example of the key secret value?

type: Opaque
----

. Create a custom resource file named `tempo.yaml` that contains the following text:
+
.Example tempo-all-in-one.yaml
[source,yaml]
----

> apiVersion: tempotracing.io/v1
> kind: Tempo
> metadata:
>   name: example

Reviewer 2: ```suggestion
apiVersion: tempo.grafana.com/v1alpha1
kind: TempoStack
metadata:
  name: simplest
spec:
  storage:
    secret:
      name: minio
      type: s3
  storageSize: 200M
  resources:
    total:
      limits:
        memory: 2Gi
        cpu: 2000m
  template:
    queryFrontend:
      jaegerQuery:
        enabled: true
        ingress:
          type: ingress
```

----

. Run the following command to deploy {TempoShortName}:
+
[source,terminal]
----
$ oc create -n tracing-system -f tempo.yaml
----

. Run the following command to watch the progress of the pods during the installation process:
+
[source,terminal]
----
$ oc get pods -n tracing-system -w
----
+
After the installation process has completed, you should see output similar to the following example:
+
[source,terminal]

> ----
> NAME                                         READY   STATUS    RESTARTS   AGE
> tempo-all-in-one-inmemory-cdff7897b-qhfdx   2/2     Running   0          24s

Reviewer 2: ```suggestion
----
NAME                                              READY   STATUS    RESTARTS   AGE
tempo-simplest2-compactor-7b9c848bb8-wn9qm        1/1     Running   0          5m11s
tempo-simplest2-distributor-657c95756d-fb27r      1/1     Running   0          5m11s
tempo-simplest2-ingester-0                        1/1     Running   0          5m11s
tempo-simplest2-querier-656d659749-7kx2f          1/1     Running   0          5m11s
tempo-simplest2-query-frontend-6bcbb97cf6-ht89j   2/2     Running   0          5m11s

```
